module/VuFind/src/VuFind/Auth/CAS.php

```diff
@@ -286,18 +287,28 @@ protected function setupCAS()
         // client can only be called once.
         if (!$this->phpCASSetup) {
             $cas = $this->getConfig()->CAS;
-            if (
-                isset($cas->log)
-                && !empty($cas->log) && isset($cas->debug) && ($cas->debug)
-            ) {
-                $casauth->setDebug($cas->log);
+            $casauth->setLogger(new PsrLoggerAdapter($this->logger));
+            if (isset($cas->debug) && ($cas->debug)) {
+                $casauth->setVerbose(true);
             }
             $protocol = constant($cas->protocol ?? 'SAML_VERSION_1_1');
+            $service_base_url = null;
+            if (isset($cas->service_base_url)) {
+                $service_base_url = $cas->service_base_url->toArray();
+            } elseif (isset($this->getConfig()->Site->url)) {
+                // fallback method
+                $service_base_url = [
+                    parse_url($this->getConfig()->Site->url, PHP_URL_SCHEME) . '://' .
```

Comment:
Is it worth calling parse_url just once and then processing the array it returns? Seems like that might be more efficient. Also, do we need to assemble this a little more carefully (e.g. checking whether some components are missing)? In particular, if there is a port number, do we need to add a ':' in front of it? I think parse_url just returns an integer in that situation.

Reply:
Another possibility might be to use a regular expression to extract the beginning of the URL, if that's easier than messing with parse_url.